### PR TITLE
Robofac trading update

### PR DESCRIPTION
**Pull request by John-Candlebury**
#### Summary

```SUMMARY: Content "Updates robofac's intercom trade list and adds caravan"```

#### Purpose of change
Helps explain how the hub keeps fed, at least initially.  Also explains why the player is not sent in missions that concern the food supply.

Also serves as an incentive for the player to live in the lab upper floor and keep working for the Hub.

#### Describe the solution
After completing the second robofac mission the caravan will spawn.

The design doc mentions that the free merchants and the old guard are not necessarily best friends forever, the merchant lampshades this by being much more dismissive of players that are also US marshals.

#### Describe alternatives you've considered
Just put some vending machines up there

#### Additional context
Camp pictures
image

### CODE REVIEW

File: data/json/mapgen/robofaq_locs/robofac_hq_chunks.json [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      { "class": "guard", "x": 7, "y": 7 },
+      { "class": "guard", "x": 10, "y": 4 }
+    ],
+    "items": { "c": { "item": "bed", "chance": 100 } },
+    "item": {
+      "r": { "item": "pillow", "chance": 33 },
+      "F": { "item": "pot", "chance": 100 },
+      "v": { "item": "jug_plastic", "chance": 100 }
+    },
+    "place_items": [ { "item": "robofac_basic_trade", "chance": 94, "x": 2, "y": [ 2, 4 ] } ]
```
Comment (John-Candlebury) [inline data/json/mapgen/robofaq_locs/robofac_hq_chunks.json]: No matter how I tell the game to spawn these items, they never spawn I have tried "items", "loot" and "place_items"

Im not sure whats causing it, Maybe its some load order related thing?

**Comment by Fris0uman** [comment]
Could they sell some of the `schematics` ? It might make some sense that they have the recipe to build robots.

**Comment by John-Candlebury** [comment]
> Could they sell some of the `schematics` ? It might make some sense that they have the recipe to build robots.

They will trade the schematics. But you will need better standing to get there.

**Comment by John-Candlebury** [comment]
Made the item spawning work through changing the item group.

Made issue #31092 to document the strange itemgroup related bug

**Comment by mlangsdorf** [comment]
Finally, one of the mystical Free Merchant caravans!  Looks good.

**Pull request merged**